libs/math/tests/math/all_tests/clustering/knn_test.cpp

```diff
@@ -36,41 +37,13 @@ TYPED_TEST_CASE(ClusteringTest, MyTypes);
 
 TYPED_TEST(ClusteringTest, knn_euclidean_test)
 {
-  using DataType  = typename TypeParam::Type;
   using ArrayType = TypeParam;
   using SizeType  = typename TypeParam::SizeType;
 
-  ArrayType A = ArrayType({4, 4});
+  ArrayType A = ArrayType::FromString("1, 2, 3, 4; 2, 3, 4, 5; -1, -2, -3, -4; -2, -3, -4, -5");
```

Comment:
Is FromString safe now? Shouldn't we use that R() stuff?

Reply:
nope - it's safe to use like this now. FromString was fixed and the raw literals aren't necessary